scripts/generate_records/generate_db_records.js

```diff
@@ -0,0 +1,387 @@
+    }
+    const executionsPerBatch = Number.parseInt(split[0], 10);
+    const granulesPerBatch = Number.parseInt(split[1], 10);
+    return { executionsPerBatch, granulesPerBatch };
+  } catch (error) {
+    throw new Error(`cannot parse ${executionsPerGranule}, expected format <executions>:<granules> ratio \n${error}`);
+  }
+};
+
+/**
+ * parse command line args for run parameters
+ *
+ * @returns {{
+ *   granules: number,
+ *   files: number,
+ *   granulesPerBatch: number
+ *   executionsPerBatch: number
+ *   collections: number
+ *   concurrency: number
+ *   variance: boolean
+ *   swallowErrors: boolean
+ * }}
+ */
+const parseArgs = () => {
+  const {
+    granulesK,
+    files,
+    executionsPerGranule,
+    collections,
+    variance,
+    concurrency,
+    swallowErrors,
+  } = minimist(
+    process.argv,
+    {
+      string: [
+        'collections',
+        'files',
+        'granulesK',
+        'executionsPerGranule',
+        'concurrency',
+      ],
+      boolean: [
+        'swallowErrors',
+        'variance',
+      ],
+      alias: {
+        g: 'granulesK',
+        f: 'files',
+        c: 'collections',
+        e: 'executionsPerGranule',
+        C: 'concurrency',
+        v: 'variance',
+        s: 'swallowErrors',
+      },
+      default: {
+        collections: process.env.COLLECTIONS || 1,
+        files: process.env.FILES || 1,
+        granulesK: process.env.GRANULES_K || 10,
+        executionsPerGranule: process.env.EXECUTIONS_PER_GRANULE || '2:2',
+        variance: process.env.VARIANCE || false,
+        concurrency: process.env.CONCURRENCY || 1,
+        swallowErrors: process.env.SWALLOW_ERRORS || true,
+      },
+    }
+  );
+  const {
+    granulesPerBatch,
+    executionsPerBatch,
+  } = parseExecutionsGranulesBatch(executionsPerGranule);
+  if (granulesPerBatch < 1) {
+    throw new Error(`granules per batch must be > 0, got ${granulesPerBatch} from ${executionsPerGranule}`);
+  }
+  if (concurrency < 1) {
+    throw new Error(`concurrency must be > 0, got ${concurrency}`);
+  }
+  return {
+    granules: Number.parseInt(granulesK, 10) * 1000,
+    files: Number.parseInt(files, 10),
+    granulesPerBatch: granulesPerBatch,
+    executionsPerBatch: executionsPerBatch,
+    collections: Number.parseInt(collections, 10),
+    concurrency: Number.parseInt(concurrency, 10),
+    variance,
+    swallowErrors,
+  };
+};
+
+/**
+ * handle command line arguments and environment variables
+ * run the data upload based on configured parameters
+ */
+const main = async () => {
+  const {
+    granules,
+    files,
+    granulesPerBatch,
+    executionsPerBatch,
+    collections,
+    variance,
+    concurrency,
+    swallowErrors,
+  } = parseArgs();
+  process.env.dbMaxPool = concurrency.toString();
+  const knex = await getKnexClient();
+  for (let collectionNumber = 0; collectionNumber < collections; collectionNumber += 1) {
```

Comment:
this doesn't work because I don't have a way to batch the outermost loop of granules within collections, I nee to loop over the collections and don't have an insert function for sub-collections, or a feasible way to make one that I've been able to come up with